src/NuGet.Services.Staging.Search/DatabaseSearchService.cs

```diff
@@ -54,10 +53,38 @@ public JObject Search(string query)
             var includePrerelease = GetIncludePrerelease(queryDictionary);
             var q = GetQuery(queryDictionary);
 
-            return _formatter.FormatSearchResults(ApplyQueryParameters(stage.Key, includePrerelease, q, skip, take));
+            return _formatter.FormatResults(SearchInternal(stage.Key, includePrerelease, q, skip, take));
         }
 
-        internal IReadOnlyList<PackageMetadata> ApplyQueryParameters(int stageKey, bool includePrerelease, string query, int skip, int take)
+        public JObject Autocomplete(string query)
+        {
+            var stage = _context.Stages.FirstOrDefault(x => x.Id == _stageId);
+
+            if (stage == null)
+            {
+                return null;
+            }
+
+            // Parse query
+            query = query.ToLowerInvariant();
+            var queryDictionary =
+                QueryHelpers.ParseNullableQuery(query)?.ToDictionary(x => x.Key, x => (string)x.Value)
+                ?? new Dictionary<string, string>();
+
+            var skip = GetSkip(queryDictionary);
+            var take = GetTake(queryDictionary);
+            var includePrerelease = GetIncludePrerelease(queryDictionary);
+            var q = GetQuery(queryDictionary);
+            var id = GetId(queryDictionary);
+
+            // Run query
+            int totalHits;
+            var result = AutocompleteInternal(stage.Key, includePrerelease, q, id, skip, take, out totalHits);
+
+            return new DatabaseAutocompleteResultsFormatter().FormatResults(result, totalHits);

```

Comment:
thanks :)

---
In reply to: [65841899](https://github.com/NuGet/NuGet.Staging/pull/58#discussion_r65841899) [](ancestors = 65841899)